libraries/chain/include/eosio/chain/wasm_eosio_injection.hpp

```diff
@@ -773,7 +777,9 @@ namespace eosio { namespace chain { namespace wasm_injections {
       }
    };
  
-   // inherit from this class and define your own injectors 
+   // "full injection" gives checktime & depth counting along with softfloat injection.
+   // Otherwise you'll just get softfloat injection
+   template<bool full_injection>
```

Comment:
It would be cleaner and more general to pass the injector(s) as template parameters.

Reply:
I thought about that but I still need to do something like
```
               if constexpr (full_injection) {
	                  wasm_ops::op_types<>::call_t chktm;
                          chktm.field = injector_utils::injected_index_mapping.find(checktime_injection::chktm_idx)->second;
                          chktm.pack(&post_code);
               }
```

Reply:
This could be moved into pre/post_injector as a function prefix (which defaults to a no-op).  That would also allow the two loops in inject to be unduplicated.  Also, while we're on this topic, why is it necessary to insert checktime both at the beginning of every function and before each function call.

Reply:
If the concern is that a boolean is not clear enough here, would an enum be better?

> Also, while we're on this topic, why is it necessary to insert checktime both at the beginning of every function and before each function call.

I can elaborate on this in private.

Reply:
It's not just the boolean.  It's more that there isn't a clean separation of responsibilities and that it's almost generic, but not quite.